### PR TITLE
#2112 Provide SL Grid Map coordinates for Official Viewer Users please

### PR DESCRIPTION
**Pull request by LLGuru**
The grid coordinates of every region are being displayed just above the region name for each tile
image

### CODE REVIEW

**Comment by akleshchev** [comment]
That seems like an odd choice of position to me. Name would be more relevant so I would have put it on top.

@kylelinden Are you fine with this?

**Comment by maxim-productengine** [comment]
It looks ok, but in my opinion, this info should be optional. We can add a simple checkbox to World map for that, or at least hide it behind debug setting (because it seems somewhat niche information).

**Comment by aiaustin** [comment]
Yes, more clutter on the maps is not needed for most people. If its needed it might be best behind a tick box with default being off.

**Comment by marchcat** [comment]
+1 for a tick box.

**Comment by LLGuru** [comment]
Comparing to the Firestorm viewer
image
image

**Comment by LLGuru** [comment]
With a check box (AKA tick box):
image
image

**Review by marchcat — COMMENTED** [review]
Is all the whitespace change in floater_world_map.xml necessary?

**Comment by Ansariel** [comment]
> Is all the whitespace change in floater_world_map.xml necessary?

If you want to annoy TPVs: yes!

**Comment by LLGuru** [comment]
> Is all the whitespace change in floater_world_map.xml necessary?

If you want to make changes safely: yes!

**Comment by LLGuru** [comment]
Using this option could help the reviewer to see the significant differences:
image

**Comment by marchcat** [comment]
> Using this option

Well, that was exactly why I'm asking.

**Comment by Ansariel** [comment]
> Using this option could help the reviewer to see the significant differences: image

Still TPVs need to merge these unnecessary changes. But in the end I'm gonna nuke those changes because our worldmap is much better anyway. 😂

**Review by marchcat — APPROVED** [review]
Ok, in this case it's a green light from me 😹

**Review by maxim-productengine — APPROVED** [review]

**Comment by akleshchev** [comment]
Please clear changes with either Vir or Kyle before merging.

> If you want to make changes safely: yes!

I don't see how that relates to safety, I feel like it's the opposite and makes conflicts likely, especially for TPVs. And it add a huge footprint to history/blame.

**Review by kylelinden — CHANGES_REQUESTED** [review]
This was implemented incorrectly. Will rewrite /issue/2112 for the desired behaviors, with a request for UX review.

**Comment by kylelinden** [comment]
> Still TPVs need to merge these unnecessary changes. But in the end I'm gonna nuke those changes because our worldmap is much better anyway. 😂

@Ansariel any chance you could give us a contribution of the FireStorm worldmap?

**Comment by Ansariel** [comment]
> > Still TPVs need to merge these unnecessary changes. But in the end I'm gonna nuke those changes because our worldmap is much better anyway. 😂
> 
> @Ansariel any chance you could give us a contribution of the FireStorm worldmap?

It's only partially my code, so I don't think I can.

**Comment by LLGuru** [comment]
Reworked according the new task definition:

1) "Include X,Y coords in the Region/Estate floater, Region tab"
image


2) "When enabled X,Y coords should appear below the region name"
image

File: indra/newview/llworldmapview.cpp [hunk 1]
```diff
@@ -517,26 +517,37 @@ void LLWorldMapView::draw()
         // Draw the region name in the lower left corner
         if (mMapScale >= DRAW_TEXT_THRESHOLD)
         {
-            std::string mesg;
+            static LLCachedControl<bool> print_coords(gSavedSettings, "MapShowGridCoords");
+
+            auto print = [&](std::string text, F32 x, F32 y)
+                {
+                    LLFontGL::getFontSansSerifSmallBold()->renderUTF8(text, 0,
```
Comment (akleshchev) [inline indra/newview/llworldmapview.cpp:524]: getFontSansSerifSmallBold() is a bit expensive, please consider saving font out of draw().

**Review by akleshchev — APPROVED** [review]

**Review by vir-linden — APPROVED** [review]
Approved

**Review by kylelinden — COMMENTED** [review]
Approved.

**Pull request merged**